### ISSUE TITLE
Broken Validation when using React Elements for Messages

### ISSUE THREAD

**Issue by dairyisscary** [react-hook-form/react-hook-form#10111]
### Version Number

7.43.2

### Codesandbox/Expo snack

n/a

### Steps to reproduce

hello react-hook-form folks!

28cafa864b4a92b2d2f6ac8f5c5cb27722e38b3f removed the option to use react elements as validation messages. as a result internationalization libraries that use elements for translation of validation messages no longer work.

For example, with `react-intl`/`formatjs`:
```tsx
<input type="text" {...form.register("name", { required: <FormattedMessage defaultMessage="this is required" /> })} />
```

most internationalization libraries (including formatjs) can produce strings but the element api is often more convenient, can use constant values, and doesn't require boilerplate to get things out of context

### Expected behaviour

hopefully we can bring this back.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
We actually only support string from the very beginning of V7, but the code was left there for awhile. Now that react hook form core is going to be framework agnostic, any react-specific code will need to be removed. Consider returning the string and using translation API if you can. eg

```tsx
required: translate(test')

// or 

required: 'test'
translate(errors.test.message)
```

**Comment by bluebill1049** [comment]
I understand this is not going to work the best for you, and certainly answering your questions will only probably bubble up more. I am trying my best to answer here with a clear mind after a whole day of work. Yes, we can consider option two, but to be mindful, this is by design to only support `string` so users don't have to type assert. The typescript never supports `React.ReactNode` in the type definition and is included in the doc. It wouldn't be my priority to fix it out probably after the library core is published and officially used by hook form.

**Comment by dairyisscary** [comment]
of course, bill! i hope i'm not being overwhelming haha (i know open source can be burnout-city).

i'll close the issue and migrate all my messages to strings :)

**Comment by bluebill1049** [comment]
@dairyisscary let me work on this later, I am moving to the core.
